lib/main.js

```diff
@@ -58,19 +58,26 @@ var init = exports.init = function(host) {
             /*
             Trigger Jenkins to build.
             */
-            if (typeof params === 'function') { 
+            if (typeof params === 'function') {
                 buildurl = build_url(BUILD, jobname);
                 callback = params;
             } else {
                 buildurl = build_url(BUILDWITHPARAMS+"?"+qs.stringify(params), jobname)
             }
-            
+
             request({method: 'POST', url: buildurl }, function(error, response) {
                 if ( error || (response.statusCode !== 201 && response.statusCode !== 302) ) {
                     callback(error, response);
                     return;
                 }
-                var data = "job is executed";
+                /*
+                Return queue location of newly-created job as per
+                  https://issues.jenkins-ci.org/browse/JENKINS-12827?focusedCommentId=201381#comment-201381
+                */
+                var data = {
+                    message: "job is executed",
+                    location: response.headers["Location"] || response.headers["location"]
+                };
                 callback(null, data);
```

Comment:
Given that `job is executed` was a completely arbitrary value for you to return, it might be better to just say 

```javascript
var data = response.headers["Location"] || response.headers["location"];
```